src/main/java/com/autotune/analyzer/experiment/ExperimentInterfaceImpl.java

```diff
@@ -91,20 +92,19 @@ public boolean addResultsToLocalStorage(Map<String, KruizeObject> mainKruizeExpe
                             deploymentObject = deploymentsMap.get(dName);
                             containersMap = deploymentObject.getContainers();
                         }
-                        List<Containers> resultContainerList = deploymentResultData.getContainers();
-                        for (Containers containers : resultContainerList) {
-                            String cName = containers.getContainer_name();
-                            String imgName = containers.getImage_name();
+                        List<ContainerObject> resultContainerObjectList = deploymentResultData.getContainerObjects();
+                        for (ContainerObject resultContainerObject : resultContainerObjectList) {
+                            String cName = resultContainerObject.getContainer_name();
                             ContainerObject containerObject;
                             if (null == containersMap.get(cName)) {
-                                containerObject = new ContainerObject(cName, imgName);
+                                containerObject = new ContainerObject();
                             } else {
                                 containerObject = containersMap.get(cName);
                             }
                             HashMap<AnalyzerConstants.AggregatorType, MetricAggregationInfoResults> aggregatorHashMap = new HashMap<>();
-                            for (AnalyzerConstants.MetricName aggregationInfoName : containers.getContainer_metrics().keySet()) {
-                                MetricAggregationInfoResults aggregatorResult = containers.getContainer_metrics().get(aggregationInfoName).get("results").getAggregationInfoResult();
-                                aggregatorHashMap.put(AnalyzerConstants.AggregatorType.valueOf(aggregationInfoName.toString()), aggregatorResult);
+                             for (ContainerMetricsHelper containerMetricsHelper : resultContainerObject.getMetrics()) {
```

Comment:
Can we get rid of `containerMetricsHelper` class. I think this can be avoided by adding the `MetricResults` directly into `ContainerObject`.  Also a more appropriate name for `ContainerObject` would be `ContainerData` as this has metrics and recommendations. It needs to be in the `common.data.result` package.

Reply:
Done